mne/annotations.py

```diff
@@ -647,6 +647,182 @@ def rename(self, mapping, verbose=None):
         return self
 
 
+class EpochAnnotationsMixin:
+    """Mixin class for Annotations in Epochs."""
+
+    @property
+    def annotations(self):  # noqa: D102
+        return self._annotations
+
+    def set_annotations(self, annotations):
+        """Setter for Epoch annotations.
+
+        Currently only works to set ``annotations`` as ``None``.
+
+        Parameters
+        ----------
+        annotations : instance of mne.Annotations | None
+            Annotations to set.
+
+        Returns
+        -------
+        self : instance of Raw
+            The raw object with annotations.
+        """
+        if annotations is not None:
+            raise RuntimeError('Setting Annotations within Epochs directly '
+                               'is not supported yet.')
```

Comment:
Why require this? I don't see why we need to restrict this. For example, if I have a raw, create epochs, go back to that raw, add some annotations, I should be able to do `epochs.set_annotations(raw.annotations)` and it should just work even with the code we have already, right?

Reply:
This was a quick fix for the tutorial bug found last week. 

For this function yes we could enable it to just receive Annotations. However, based on raw_sfreq being set correctly, this could be misconstrued? E.g. if someone sets up a epochsKit file that was resampled before Annotation support in Epochs. 

Should I just document a warning in the doc string or should I explicitly have a warning if they are one of the classes that I hard code raw_sfreq?

Reply:
I removed this error and added some notes to the doc string.